### PR TITLE
Do not upgrade dependencies on v1-10-stable

### PR DESCRIPTION
**Pull request by turbaszek**
We should not upgrade requirements version on v1-10-stable branch
as it seems to be failing currently:
https://github.com/apache/airflow/runs/1316476959

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -31,7 +31,7 @@ env:
   SKIP_CI_IMAGE_CHECK: "true"
   DB_RESET: "true"
   VERBOSE: "true"
-  UPGRADE_TO_LATEST_CONSTRAINTS: ${{ github.event_name == 'push' || github.event_name == 'scheduled' }}
+  UPGRADE_TO_LATEST_CONSTRAINTS: ${{ github.event_name == 'scheduled' }}
```
Comment (potiuk) [inline .github/workflows/ci.yml]: We still want to upgrade those on master push.

Comment (potiuk) [inline .github/workflows/ci.yml]:
```suggestion
  UPGRADE_TO_LATEST_CONSTRAINTS: ${{ github.event_name == 'scheduled' || (github.event_name == 'push' && github.base_ref == 'master') }}
```

**Comment by potiuk** [comment]
too long line :(

**Comment by potiuk** [comment]
And I think there is a problem with master vs. stable version of build scripts now - the images (for security are always build using master ci scripts and I think there was a backwards -incompatible change . But it will be easy to fix

**Comment by potiuk** [comment]
@turbaszek - add this to `scripts/ci/libriaries/_initialisation.sh` (see where it is added in master). This should help with image building.

```
# Determine current branch
function initialization::initialize_branch_variables() {
    # Default branch used - this will be different in different branches
    export DEFAULT_BRANCH=${DEFAULT_BRANCH="v1-10-test"}
    export DEFAULT_CONSTRAINTS_BRANCH=${DEFAULT_CONSTRAINTS_BRANCH="constraints-1-10"}
    readonly DEFAULT_BRANCH
    readonly DEFAULT_CONSTRAINTS_BRANCH

    # Default branch name for triggered builds is the one configured in default branch
    # We need to read it here as it comes from _common_values.sh
    export BRANCH_NAME=${BRANCH_NAME:=${DEFAULT_BRANCH}}
}
```

Explanation: 

Since we cannot 'source' the scripts coming from the PR (it's dangerous because we could run malicious code) we grep _initialization.sh to find out which is the image branch and what is the constraints branch.

This is done (and explained) here: https://github.com/apache/airflow/blob/db121f726b3c7a37aca1ea05eb4714f884456005/.github/workflows/build-images-workflow-run.yml#L280

**Comment by potiuk** [comment]
@turbaszek -> here the problem is slightly more complex. What you need to do is two things:

1) In Dockerfile.ci:

* move scripts/ci/in_container/entrypoint_ci.sh  to  scripts/in_container/entrypoint_ci.sh
* replace the failing COPY line in Dockerfile.ci with

```
COPY scripts/in_container/entrypoint_ci.sh /entrypoint
```

2) In Dockerfile

* move scripts/ci/in_container/entrypoint_prod.sh and clean-logs.sh,  to scripts/in_container/prod

* Replace the two lines in Dockerfile to:

```
COPY --chown=airflow:root scripts/in_container/prod/entrypoint_prod.sh /entrypoint
COPY --chown=airflow:root scripts/in_container/prod/clean-logs.sh /clean-logs
```

That should fix the current failure.

Sorry for that, but the opitimization of workflow_run and keeping it secure makes it rather difficult to avoid such problems from time to time. It's just bad timing.

**Comment by turbaszek** [comment]
@potiuk do you have any idea what is happening with docs build?
```
docker: Error response from daemon: OCI runtime create failed: container_linux.go:370: starting container process caused: process_linux.go:459: container init caused: rootfs_linux.go:59: mounting "/home/runner/work/airflow/airflow/scripts/ci/in_container/entrypoint_ci.sh" to rootfs at "/var/lib/docker/overlay2/12e5fca5cad13cc8c3c856907976241376b1d6af9d0195918753a1351cde22cd/merged/entrypoint" caused: not a directory: unknown: Are you trying to mount a directory onto a file (or vice-versa)? Check if the specified host path exists and is the expected type.
70
time="2020-11-02T18:44:30Z" level=error msg="error waiting for container: context canceled"
```

**Comment by potiuk** [comment]
I pushed a fixup that should fix it. It was still reference to the old "in_container" script in docker compose files - where this file has moved to "in_container" from "ci/in_container".

**Comment by potiuk** [comment]
Seems t works :)

**Review by potiuk — APPROVED** [review]

**Pull request merged**